File: src/main/java/skywolf46/extrautility/util/PageUtility.java
package skywolf46.extrautility.util;

import java.util.ArrayList;
import java.util.List;

public class PageUtility {
    public static <T> int pages(List<T> list, int itemPerPage) {
        if (list.size() == 0)
            return 0;
        if (list.size() <= itemPerPage)
            return 1;
        return list.size() / itemPerPage + (list.size() % itemPerPage == 0 ? 0 : 1);
    }

    public static <T> List<T> splitPage(List<T> list, int itemPerPage, int page) {
        int listPage = pages(list, itemPerPage);
        if (listPage < page)
            return new ArrayList<>();
        return list.subList((page - 1) * itemPerPage, Math.min(list.size(), page * itemPerPage));
    }
}

File: common/src/main/kotlin/skywolf46/extrautility/util/PageUtility.kt
package skywolf46.extrautility.util

import java.util.ArrayList
import kotlin.math.min


object PageUtility {
    fun <T> pages(list: List<T>, itemPerPage: Int): Int {
        if (list.isEmpty()) return 0
        return if (list.size <= itemPerPage) 1 else list.size / itemPerPage + if (list.size % itemPerPage == 0) 0 else 1
    }

    fun <T> splitPage(list: List<T>, itemPerPage: Int, page: Int): List<T> {
        val listPage = pages(list, itemPerPage)
        return if (listPage < page) ArrayList() else list.subList(
            (page - 1) * itemPerPage,
            min(list.size, page * itemPerPage)
        )
    }
}